pkg/dns/provider/interface.go

```diff
@@ -23,19 +23,20 @@ import (
 )
 
 type Config struct {
-	TTL                      int64
-	CacheTTL                 time.Duration
-	RescheduleDelay          time.Duration
-	StatusCheckPeriod        time.Duration
-	Ident                    string
-	Dryrun                   bool
-	ZoneStateCaching         bool
-	DisableDNSNameValidation bool
-	Delay                    time.Duration
-	EnabledTypes             utils.StringSet
-	Options                  *FactoryOptions
-	Factory                  DNSHandlerFactory
-	RemoteAccessConfig       *embed.RemoteAccessServerConfig
+	TTL                                         int64
+	CacheTTL                                    time.Duration
+	RescheduleDelay                             time.Duration
+	StatusCheckPeriod                           time.Duration
+	Ident                                       string
+	Dryrun                                      bool
+	ZoneStateCaching                            bool
+	DisableDNSNameValidation                    bool
+	Delay                                       time.Duration
+	EnabledTypes                                utils.StringSet
+	Options                                     *FactoryOptions
+	Factory                                     DNSHandlerFactory
+	RemoteAccessConfig                          *embed.RemoteAccessServerConfig
+	MaxMetadataRecordDeletionsPerReconciliation int
```

Comment:
Well, this field will only exist for one or two releases and is only used in a few places.
Suggestions for a shorter name without loss of much information are welcome!

Reply:
I didn't mean to criticize the length of the field name :) I think it's totally fair and appropriate :D As otherwise stated, I usually prefer verbose, clear names but I'll take any chance that I can get to bring up my favorite Welsh train station 🚂